### DISCUSSION TITLE
Is it possible to assign env variables per accessed domain?

### DISCUSSION THREAD

**Discussion by zomars** [vercel/vercel#8374] [Help] [answered]
Let's say we have a project with `www.example.com` and `nightly.example.com` as production domains. Is there a way to change the `DATABASE_URL` if accessed from `nightly.example.com` vs `www.example.com` ?

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @zomars! I don't know of any way to set up domain-specific variables in the Dashboard, but you can create this behavior in your code. For example, you could check the `host` and direct traffic to the `DATABASE_URL` or `DATABASE_URL_NIGHTLY` like
```
export async function getServerSideProps(context) {
  const host = context.req.headers.host;
  let databaseUrl = process.env.DATABASE_URL;
  if (domain === "nightly.example.com") {
    // Use nightly database url
    databaseUrl = process.env.DATABASE_URL_NIGHTLY;
  }

  // Get data from db

  return {
    props: {
      data,
    },
  };
}
```

**Reply by zomars** [reply]
Is there a way to overwrite the env altogether in `next.config.js` or that wouldn't work?

**Reply by amyegan** [reply]
You should take a look at the Next.js documentation on [environment variables](https://nextjs.org/docs/basic-features/environment-variables) and [env var configuration](https://nextjs.org/docs/api-reference/next.config.js/environment-variables). I think that will help you achieve what you're trying to do.

**Reply by basememara** [reply]
@zomars have you figured this out? Customizing and stacking env vars is easy, but when you need to base it on something dynamic like a hostname it becomes tricky.

**Reply by subproject22** [reply]
Vercel Env Variables are brain dead. Can't believe it's 2024 and still no way to differ between domains with variable. Why the hell you offer multiple domains per project in first place then?

**Reply by amyegan** [reply]
@subproject22 I understand you're frustrated, but please don't be rude. 

The original post is a few years old and some things have changed since then. I recommend looking into the available [System Environment Variables](https://vercel.com/docs/projects/environment-variables/system-environment-variables#system-environment-variables). `VERCEL_URL`, `VERCEL_BRANCH_URL`, or `VERCEL_PROJECT_PRODUCTION_URL` may be what you're looking for.
